decimal.go

```diff
@@ -4,6 +4,10 @@ import (
 	"io"
```

Comment:
Nice work! Do we need to do anything special for marshaling `BinaryTreeReader` or `DateTreeReader`?

Reply:
Just checked both and they marshal correctly as they are now. `BinaryTreeReader` just returns a slice of bytes that then becomes a BSON `$binary` value, and `DateTimeReader` is an embedded struct that contains a `time.Time` value but marshals into the correct value:

'{"$date":{"$numberLong":"1577997435762"}}'

Reply:
Nice.